### PR TITLE
Feature/logout callback

### PR DESCRIPTION
**Pull request by mcjustin**

### CODE REVIEW

File: OidcStrategy.php [hunk 1]
```diff
@@ -58,6 +59,7 @@ public function request(){
      * Internal callback; handle response to authorization request and request new access token
      */
     public function oauth2callback(){
+        CakeLog::write(LOG_DEBUG, __CLASS__."->".__FUNCTION__."(), here's what's in _GET:" . print_r($_GET, true) . ", here's what in the request headers:" . print_r(apache_request_headers(), true));
```
Comment (ivan-c) [inline OidcStrategy.php]:
```suggestion
        // CakeLog::write(LOG_DEBUG, __CLASS__."->".__FUNCTION__."(), here's what's in _GET:" . print_r($_GET, true) . ", here's what in the request headers:" . print_r(apache_request_headers(), true));
```

File: OidcStrategy.php [hunk 2]
```diff
@@ -136,6 +138,57 @@ public function oauth2callback(){
...
+        $userObj = new User();
+        $user = $userObj->find('first', array('conditions' => array('User.external_id' => $sub),
+            'recursive' => -1));
+        //CakeLog::write(LOG_DEBUG, __CLASS__."->".__FUNCTION__."(), mapped sub $sub to user:" . print_r($user, true));
+        $userId = $user['User']['id'];
+
+        $sessionObj = new DatabaseSessionPlusUserId();
+        $deleteResult = $sessionObj->deleteByUserId($userId);
+
+        CakeLog::write(LOG_DEBUG, __CLASS__ ."->". __FUNCTION__ . "(), done.");
```
Comment (ivan-c) [inline OidcStrategy.php]:
```suggestion
        // CakeLog::write(LOG_DEBUG, __CLASS__ ."->". __FUNCTION__ . "(), done.");
        CakeLog::write(LOG_DEBUG, "logged out user $userId by OIDC back-channel logout");
```

File: OidcStrategy.php [hunk 3]
```diff
@@ -136,6 +138,57 @@ public function oauth2callback(){
...
+        $sessionCount = $sessionObj->find('count',
+            array('conditions' => array('DatabaseSessionPlusUserId.user_id' => $dhair2UserId)));
+        $returnVal = false;
+        if ($sessionCount >= 1) $returnVal = true;
+
+        CakeLog::write(LOG_DEBUG, __CLASS__ ."->". __FUNCTION__ . "($dhair2UserId), returning $returnVal.");
+        return $returnVal;
+    }
+     */
+
```
Comment (ivan-c) [inline OidcStrategy.php]:
```suggestion
```

**Review by ivan-c — APPROVED** [review]
Looks good! Had a few logging suggestions but ready to merge

**Comment by mcjustin** [comment]
Corresponding change to dhair2 `users` table schema: https://gitlab.cirg.washington.edu/svn/dhair2/-/merge_requests/780

**Comment by mcjustin** [comment]
See also https://gitlab.cirg.washington.edu/svn/dhair2/-/merge_requests/781

**Pull request merged**